fsspec/spec.py

```diff
@@ -8,75 +10,176 @@
 
 logger = logging.getLogger("fsspec")
 
-# alternative names for some methods, which get patched to new instances
-# (alias, original)
-aliases = [
-    ("makedir", "mkdir"),
-    ("mkdirs", "makedirs"),
-    ("listdir", "ls"),
-    ("cp", "copy"),
-    ("move", "mv"),
-    ("stat", "info"),
-    ("disk_usage", "du"),
-    ("rename", "mv"),
-    ("delete", "rm"),
-    ("upload", "put"),
-    ("download", "get"),
-]
 
-try:  # optionally derive from pyarrow's FileSystem, if available
-    import pyarrow as pa
+def _warn_implicit_alias(method):
+    @functools.wraps(method)
+    def wrapper(*args, **kwargs):
+        self = args[0]
+        if not self._add_aliases:
+            warnings.warn(
+                self._alias_message.format(method.__name__), FutureWarning, stacklevel=2
+            )
+        return method(*args, **kwargs)
 
-    up = pa.filesystem.DaskFileSystem
-except ImportError:
-    up = object
+    return wrapper
+
+
+class AliasMixin:
+    """
+    Mixin providing aliases for common methods.
+
+    Notes
+    -----
+    By default this is included for all sublasses inheriting from AbstractFileSystem, but this
+    behavior is deprecated. In the future, inheriting from AliasMixin will be required to use
+    these aliases.
+
+    In the interim, using an implicitly inherited alias will emit a warning. To silence the warning,
+    inherit from AliasMixin and set `_add_aliases=True` on the filesystem class.
+    """
+
+    _add_aliases = False
+    _alias_message = "Using implicitly added alias '{}'. Inherit from AliasMixin to continue using this method."
+
+    @_warn_implicit_alias
+    def makedir(self, path, create_parents=True, **kwargs):
```

Comment:
Might we not list the explicit signatures, since then we have to change them in multiple places in the future? 
Also the repetitive pattern might be explicit, but it is verbose and perhaps error-prone.

Reply:
This preserves the function signature. If you want to use you certainly could just have `*args, **kwargs`, but then you would need more magic to rewrite the signature.